src/bridge/cockpitpackages.c

```diff
@@ -793,9 +801,17 @@ handle_package_manifests_js (CockpitWebServer *server,
 
   out_headers = cockpit_web_server_new_table ();
 
-  if (!packages->checksum)
-    cockpit_web_response_set_cache_type (response, COCKPIT_WEB_RESPONSE_NO_CACHE);
-
+  if (packages->checksum)
+    {
+      g_hash_table_insert (out_headers, g_strdup (CHECKSUM_HEADER),
+                           g_strdup (packages->checksum));
+      g_hash_table_insert (out_headers, g_strdup ("ETag"),
+                           g_strdup_printf ("\"$%s\"", packages->checksum));
```

Comment:
Generally the etag is set by cockpit-ws in channel response. Based on if a checksum url was used and the language requests. However, the javascript parses the checksum out of the etag after calling the manifest file, so it's important that this one always has the etag. By setting it here for the manifest urls we can ensure that the manifests responses always get the etag they need without having to set it here for all the requests